sequential/backend/build_mdd.cpp

```diff
@@ -8,7 +8,7 @@
 #include "build_mdd.hpp"
 
 //Populates the MDD node vector
-void Popl_nodes(int N, int L, int num_att,vector<int>* max_attrs, vector<int>* min_attrs,
+void Popl_nodes(vector<Node*>* datab_m, vector<Pattern*>* dfs_q, int N, int L, int num_att,vector<int>* max_attrs, vector<int>* min_attrs,
```

Comment:
curiosity question; are you passing these data structures to save on memory? 

If I use the same seq2pat object twice, say I mine once, and then mine again. would this be problem that it's passed by reference here? Can we test plz that repeated calls to seq2pat work ok, if not tested already. Same for remove a constraint and then running again the same seq2pat model

Reply:
Added a test for the repeated calls to seq2pat, it works fine. Also added a test to call seq2pat without constraint, then add constraint, then remove constraint, the results look fine after each call.

Reply:
nice!